axonserver/src/main/java/io/axoniq/axonserver/access/user/UserController.java

```diff
@@ -45,14 +55,37 @@ public void deleteUser(String username) {
         }
     }
 
+    /**
+     * Return a list of all users.
+     *
+     * @return a {@link List} of all {@link User}s.
+     */
     public List<User> getUsers() {
         return userRepository.findAll();
     }
 
-    private User syncUser(String username, String password, Set<UserRole> roles) {
+    /**
+     * Updates/creates a user with specified password and roles. If password is set, it will be hashed. If no password
+     * is set and the user already exists the stored password is not changed. If the user does not exist, and the
+     * password is {@code null}, the user will get a {@link #PWD_NOLOGON} marker instead.
+     *
+     * Note that the {@link #PWD_NOLOGON} marker prevents a match with any password, as it is not hashed.
+     *
+     * @param username the username
+     * @param password the plaintext password
+     * @param roles    the roles granted to the user
+     * @return the stored information, including the current (hashed) password.
+     */
+    @Transactional
+    public User updateUser(String username, String password, Set<UserRole> roles) {
         synchronized (userRepository) {
             if (StringUtils.isEmpty(password)) {
                 password = getPassword(username);
+                if (password == null) {
```

Comment:
password == PWD_NOLOGON?

Reply:
As discussed: getPassword(username) will only return null if the user is not found. Will change to use findUser(username) to make this more explicit.